src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
+
+  let sum = 0;
+
+  children.forEach((ch) => sum += ch.born - people.find(
+    (m) => m.name === ch.mother
+  ).born);
+
+  return sum / children.length;
```

Comment:
This function is incorrect, you need to change the formatting and rewrite this using reduce function, similarly to previous examples: 

```
function calculateAverageAgeDiff(people, onlyWithSon) {
    const children = onlyWithSon
      ? people.filter(
          (p) => p.sex === "m" && people.some((m) => m.name === p.mother)
        )
      : people.filter((p) => people.some((m) => m.name === p.mother));

    const result = children.reduce((accumulator, child) => {
      const mother = people.find((m) => m.name === child.mother);
      if (mother) {
        return accumulator + (child.born - mother.born);
      } else {
        return accumulator;
      }
    }, 0);

    return result / children.length;
  }
```